cpp/src/arrow/dataset/file_parquet_test.cc

```diff
@@ -60,42 +60,43 @@ Status WriteRecordBatch(const RecordBatch& batch, FileWriter* writer) {
   return Status::OK();
 }
 
-Status WriteRecordBatchReader(RecordBatchReader* reader, FileWriter* writer) {
-  auto schema = reader->schema();
+Status WriteRecordBatchReader(RecordBatchReader&& reader, FileWriter* writer) {
```

Comment:
Is there a reason for making these changes or is it just for aesthetics?

Reply:
I was making it more clear from the function signature that the RecordBatchReader is consumed by WriteRecordBatchReader. Aesthetics, I suppose

Reply:
I don't know, I find it a bit confusing to be frank. We use `RecordBatchReader*` to point out that it's a non-const reference/pointer argument.

Reply:
alright, will do